src/sql/base/browser/ui/table/plugins/autoSizeColumns.plugin.ts

```diff
@@ -134,27 +149,78 @@ export class AutoColumnSize<T extends Slick.SlickData> implements Slick.Plugin<T
 		}
 	}
 
+	/**
+	 * For each column, find the max width of the texts in the first 10 rows.
+	 * @param columnDefs Column definitions of all columns that need to be resized
+	 * @param colIndices Column indices of all columns that need to be resized
+	 * @returns An array of the max widths of each column
+	 */
+	private getMaxColumnTextWidths(columnDefs: Slick.Column<T>[], colIndices: number[]): number[] {
+		let data = this._grid.getData() as Slick.DataProvider<T>;
+		let viewPort = this._grid.getViewport();
+		let start = Math.max(0, viewPort.top);
+		let end = Math.min(data.getLength(), MAX_ROWS_TO_SCAN);
+		let allTexts: Array<string>[] = [];
+		let rowElements: JQuery[] = [];
+
+		columnDefs.forEach((columnDef) => {
+			let texts: Array<string> = [];
+			for (let i = start; i < end; i++) {
+				texts.push(data.getItem(i)[columnDef.field!]);
+			}
+			allTexts.push(texts);
+			let rowEl = this.createRow();
+			rowElements.push(rowEl);
+		});
+
+		let templates = this.getMaxTextTemplates(allTexts, columnDefs, colIndices, data, rowElements);
+
+		let widths = this.getTemplateWidths(rowElements, templates);
+
+		return widths.map((width) => {
+			return width > this._options.maxWidth! ? this._options.maxWidth! : width;
+		});
+	}
+
 	private getMaxColumnTextWidth(columnDef: Slick.Column<T>, colIndex: number): number {
 		let texts: Array<string> = [];
 		let rowEl = this.createRow();
 		let data = this._grid.getData() as Slick.DataProvider<T>;
 		let viewPort = this._grid.getViewport();
 		let start = Math.max(0, viewPort.top);
-		let end = Math.min(data.getLength(), viewPort.bottom);
+		let end = Math.min(data.getLength(), MAX_ROWS_TO_SCAN);
 		for (let i = start; i < end; i++) {
 			texts.push(data.getItem(i)[columnDef.field!]);
 		}
 		let template = this.getMaxTextTemplate(texts, columnDef, colIndex, data, rowEl);
-		let width = this.getTemplateWidth(rowEl, template);
+		let width = this.getTemplateWidths([rowEl], [template])[0];
 		this.deleteRow(rowEl);
 		return width > this._options.maxWidth! ? this._options.maxWidth! : width;
 	}
 
-	private getTemplateWidth(rowEl: JQuery, template: JQuery | HTMLElement | string): number {
-		let cell = jQuery(rowEl.find('.slick-cell'));
-		cell.append(template);
-		jQuery(cell).find('*').css('position', 'relative');
-		return cell.outerWidth() + 1;
+	private getTemplateWidths(rowElements: JQuery[], templates: (JQuery | HTMLElement | string)[]): number[] {
+		let cells: JQuery[] = [];
+
+		// Write all changes first then read all widths to prevent layout thrashing
+		// (https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing)
+		templates.forEach((template, index) => {
+			let rowEl = rowElements[index];
+			let cell = jQuery(rowEl.find('.slick-cell'));
+			cell.append(template);
+			jQuery(cell).find('*').css('position', 'relative');
+			cells.push(cell);
+		});
+
+		return cells.map(cell => cell.outerWidth() + 1);
```

Comment:
Add comment explaining why + 1

Reply:
Not sure what the exact reason is but taking out the + 1 results in the width not being large enough for some header texts. Might be because of padding or border css for the grids.
![Screen Shot 2021-08-12 at 10 44 28 AM](https://user-images.githubusercontent.com/23462877/129245070-ce1e85fe-8620-4a26-9725-6bf41f1816e3.png)